keras_cv/models/task.py

```diff
@@ -0,0 +1,166 @@
+        # models is nested and cannot be passed to our Task constructors.
+        return {
+            "backbone": keras.layers.serialize(self.backbone),
+            "name": self.name,
+            "trainable": self.trainable,
+        }
+
+    @classmethod
+    def from_config(cls, config):
+        # The default `from_config()` for functional models will return a
+        # vanilla `keras.Model`. We override it to get a subclass instance back.
+        if "backbone" in config and isinstance(config["backbone"], dict):
+            config["backbone"] = keras.layers.deserialize(config["backbone"])
+        return cls(**config)
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return {}
+
+    @classproperty
+    def presets_with_weights(cls):
+        """Dictionary of preset names and configurations that include weights."""
+        return {}
+
+    @classproperty
+    def backbone_presets(cls):
+        """Dictionary of preset names and configurations for compatible backbones."""
+        return {}
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        load_weights=None,
+        **kwargs,
+    ):
+        """Instantiate {{model_name}} model from preset architecture and weights.
+
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+                If looking for a preset with pretrained weights, choose one of
+                "{{preset_with_weights_names}}".
+            load_weights: Whether to load pre-trained weights into model.
+                Defaults to `None`, which follows whether the preset has
+                pretrained weights available.
+
+        Examples:
+        ```python
+        # Load architecture and weights from preset
+        model = keras_cv.models.{{model_name}}.from_preset(
+            "{{example_preset_name}}",
+        )
+
+        # Load randomly initialized model from preset architecture with weights
+        model = keras_cv.models.{{model_name}}.from_preset(
+            "{{example_preset_name}}",
+            load_weights=False,
+        ```
+        """
+
+        if not cls.presets:
+            raise NotImplementedError(
+                "No presets have been created for this class."
+            )
+
+        if preset not in cls.presets:
+            raise ValueError(
+                "`preset` must be one of "
+                f"""{", ".join(cls.presets)}. Received: {preset}."""
+            )
+
+        metadata = cls.presets[preset]
+
+        # Check if preset is backbone-only model
+        if preset in cls.backbone_presets:
+            backbone = keras.utils.deserialize_keras_object(metadata)
+            return cls(backbone, **kwargs)
+
+        # Otherwise must be one of class presets
+        config = metadata["config"]
+        model = cls.from_config({**config, **kwargs})
+
+        if not load_weights:
```

Comment:
The docstring suggests that the default `None` value for `load_weights` will follow whether the preset has weights, but this implementation is inconsistent.

(so right now, no weight loading is happening unless `load_weights=True`

I'm working on migrating the checkpoints now.

Reply:
Thanks, looks like a bad merge actually!

Reply:
Fixed